Platform/ApollolakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -589,6 +589,25 @@ ClearFspHob (
   LdrGlobal->FspHobList = NULL;
 }
 
+/**
+  Platform specific initialization for BSP and APs.
+**/
+VOID
+EFIAPI
+PlatformCpuInit (
```

Comment:
Need more documentation on what the limitations are for this function. Basically somethings like - no access to shared resource unless acquired by a lock, etc.

Reply:
Let me use spin lock before and after this function.

Reply:
I don't think we need to add lock by default.   We need to add it when it is required.   In this case for APL it is not required.   Lock will block parallel execution for all threads and will slow down the MpInit flow.

Reply:
OK. let me check it back and we could add spin lock when needed later.